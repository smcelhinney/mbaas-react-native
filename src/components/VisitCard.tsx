import * as React from 'react';
import { Component } from 'react';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  ScrollView,
  Dimensions
} from 'react-native';
import {
  sharedTypographyStyles,
  colorScheme,
  sharedElementStyles
} from '../constants/styles';
import { graphql } from 'react-apollo';
import { format, distanceInWordsToNow } from 'date-fns';

const DEVICE_WIDTH = Dimensions.get('window').width;

interface VisitCardProps {
  id: string;
  visitDate: string;
  visitLocation: string;
  infections: any;
  antibiotics: any;
  modifiedAt: string;
}

export default class VisitCard extends Component<VisitCardProps, {}> {
  render() {
    const {
      id,
      visitDate,
      visitLocation,
      infections,
      antibiotics,
      modifiedAt
    } = this.props;
    return (
      <View style={styles.visitCard} key={id}>
        <View style={styles.visitCardHeader}>
          <Text style={styles.cardSubHeaderText}>
            {format(visitDate, 'Do MMMM YYYY').toUpperCase()}{' '}
          </Text>
          <Text style={styles.cardHeaderText}>
            {visitLocation || 'Location not provided'}{' '}
          </Text>
        </View>
        <View style={styles.visitCardContent}>
          <Text style={styles.cardSubHeaderText}>INFECTIONS</Text>
          <View
            style={{
              borderLeftWidth: 4,
              borderColor: '#4395BF',
              marginTop: 0,
              marginBottom: 15,
              paddingLeft: 10
            }}
          >
            {this.renderSummary(infections.edges, 'infections')}
          </View>
          <Text style={styles.cardSubHeaderText}>ANTIBIOTICS</Text>
          <View
            style={{
              borderLeftWidth: 4,
              borderColor: '#79C8A6',
              marginTop: 0,
              marginBottom: 15,
              paddingLeft: 10
            }}
          >
            {this.renderSummary(antibiotics.edges, 'antibiotics')}
          </View>
        </View>
      </View>
    );
  }

  private renderMore(edges: any) {
    return edges.length > 2 ? (
      <Text style={[styles.cardHeaderText, styles.summaryText]}>
        {'+ ' + (edges.length - 2) + ' more'}
      </Text>
    ) : null;
  }

  private renderSummary(edges: any, type: string) {
    const hasReferenceItems = edges.length > 0;
    return hasReferenceItems ? (
      <View>
        {edges.map((edge: any, index: number) => {
          if (index === 0 || index === 1) {
            return (
              <Text
                key={edge.node.id}
                style={[styles.cardHeaderText, styles.summaryText]}
              >
                {edge.node.name}
              </Text>
            );
          }
        })}
        {this.renderMore(edges)}
      </View>
    ) : (
      <Text style={[styles.cardHeaderText, styles.summaryText]}>
        No {type} recorded
      </Text>
    );
  }
}

const styles: any = StyleSheet.create({
  ...sharedTypographyStyles,
  visitCard: sharedElementStyles.visitCard,
  visitCardContent: {
    flex: 1,
    justifyContent: 'flex-start',
    padding: 15
  },
  listMetaCardContent: {
    flex: 1
  },
  visitCardHeader: {
    justifyContent: 'center',
    height: 60,
    borderBottomWidth: 1,
    borderBottomColor: colorScheme.lightGrey,
    paddingHorizontal: 20
  },
  visitCardFooter: {
    height: 0,
    alignItems: 'center',
    justifyContent: 'center'
  },
  summaryText: {
    color: colorScheme.darkGrey,
    paddingVertical: 3
  }
});
