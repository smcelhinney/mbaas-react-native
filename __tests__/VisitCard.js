import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import VisitCard from '../build/components/VisitCard.js';
import toJson from 'enzyme-to-json';

describe('Component: VisitCard', () => {

  const testEdges = [
    {
      node: {
        id: "001",
        name: "test reference item 1"
      }
    },
    {
      node: {
        id: "002",
        name: "test reference item 2"
      }
    }
  ]

  const props = {
    infections: {
      edges: testEdges,
    },
    antibiotics: {
      edges: testEdges,
    }
  }

  it('renders correctly', () => {
    const wrapper = shallow(
      <VisitCard {...props}/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  })

})