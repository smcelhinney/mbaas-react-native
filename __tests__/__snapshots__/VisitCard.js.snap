// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: VisitCard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "height": 300,
      "marginRight": 15,
      "paddingBottom": 10,
      "shadowColor": "#000000",
      "shadowOpacity": 0.1,
      "shadowRadius": 5,
      "width": 690,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EFEFF4",
        "borderBottomWidth": 1,
        "height": 60,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#8e8e93",
          "fontFamily": "Helvetica Neue",
          "fontSize": 11,
          "fontWeight": "bold",
          "marginBottom": 7,
        }
      }
    >
      INVALID DATE
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#82C5E6",
          "fontFamily": "Helvetica Neue",
          "fontSize": 16,
        }
      }
    >
      Location not provided
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "padding": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#8e8e93",
          "fontFamily": "Helvetica Neue",
          "fontSize": 11,
          "fontWeight": "bold",
          "marginBottom": 7,
        }
      }
    >
      INFECTIONS
    </Text>
    <View
      style={
        Object {
          "borderColor": "#4395BF",
          "borderLeftWidth": 4,
          "marginBottom": 15,
          "marginTop": 0,
          "paddingLeft": 10,
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          key="001"
          style={
            Array [
              Object {
                "color": "#82C5E6",
                "fontFamily": "Helvetica Neue",
                "fontSize": 16,
              },
              Object {
                "color": "#606060",
                "paddingVertical": 3,
              },
            ]
          }
        >
          test reference item 1
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          key="002"
          style={
            Array [
              Object {
                "color": "#82C5E6",
                "fontFamily": "Helvetica Neue",
                "fontSize": 16,
              },
              Object {
                "color": "#606060",
                "paddingVertical": 3,
              },
            ]
          }
        >
          test reference item 2
        </Text>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#8e8e93",
          "fontFamily": "Helvetica Neue",
          "fontSize": 11,
          "fontWeight": "bold",
          "marginBottom": 7,
        }
      }
    >
      ANTIBIOTICS
    </Text>
    <View
      style={
        Object {
          "borderColor": "#79C8A6",
          "borderLeftWidth": 4,
          "marginBottom": 15,
          "marginTop": 0,
          "paddingLeft": 10,
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          key="001"
          style={
            Array [
              Object {
                "color": "#82C5E6",
                "fontFamily": "Helvetica Neue",
                "fontSize": 16,
              },
              Object {
                "color": "#606060",
                "paddingVertical": 3,
              },
            ]
          }
        >
          test reference item 1
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          key="002"
          style={
            Array [
              Object {
                "color": "#82C5E6",
                "fontFamily": "Helvetica Neue",
                "fontSize": 16,
              },
              Object {
                "color": "#606060",
                "paddingVertical": 3,
              },
            ]
          }
        >
          test reference item 2
        </Text>
      </View>
    </View>
  </View>
</View>
`;
